test/unit/parser-test.js

```diff
@@ -0,0 +1,122 @@
+define([
+    'utils/parser'
+], function (parser) {
+    /* jshint qunit: true */
+
+    module('parser');
+
+    var testerGenerator = function (assert, method) {
+        return function (left, right, message) {
+            assert.strictEqual(method.apply(this, left), right, message);
+        };
+    };
+
+    test('parser.getAbsolutePath', function(assert) {
+        var path = parser.getAbsolutePath(null, null);
+        assert.notOk(path);
+
+        path = parser.getAbsolutePath('https://testingUrl', null);
+        assert.equal(path, 'https://testingUrl');
+
+        path = parser.getAbsolutePath('path', 'base');
+        assert.ok(path.indexOf('path') >= 0);
+        assert.ok(path.indexOf('base') >= 0);
+
+        var test = testerGenerator(assert, parser.getAbsolutePath);
+        test(['.',   'https://example.com/alpha/beta/gamma'], 'https://example.com/alpha/beta');
```

Comment:
gamma looks like a directory, but is treated as a file. Maybe naming it that way in the test would make the test clearer.